ts/src/deribit.ts

```diff
@@ -709,114 +709,125 @@ export default class deribit extends Exchange {
          * @param {object} [params] extra parameters specific to the exchange API endpoint
          * @returns {object[]} an array of objects representing market data
          */
-        const currenciesResponse = await this.publicGetGetCurrencies (params);
-        //
-        //     {
-        //         "jsonrpc": "2.0",
-        //         "result": [
-        //             {
-        //                 "withdrawal_priorities": [
-        //                     { value: 0.15, name: "very_low" },
-        //                     { value: 1.5, name: "very_high" },
-        //                 ],
-        //                 "withdrawal_fee": 0.0005,
-        //                 "min_withdrawal_fee": 0.0005,
-        //                 "min_confirmations": 1,
-        //                 "fee_precision": 4,
-        //                 "currency_long": "Bitcoin",
-        //                 "currency": "BTC",
-        //                 "coin_type": "BITCOIN"
-        //             }
-        //         ],
-        //         "usIn": 1583761588590479,
-        //         "usOut": 1583761588590544,
-        //         "usDiff": 65,
-        //         "testnet": false
-        //     }
-        //
-        const parsedMarkets = {};
-        const currenciesResult = this.safeValue (currenciesResponse, 'result', []);
+        const instrumentsResponses = [];
         const result = [];
-        for (let i = 0; i < currenciesResult.length; i++) {
-            const currencyId = this.safeString (currenciesResult[i], 'currency');
-            const request = {
-                'currency': currencyId,
-            };
-            const instrumentsResponse = await this.publicGetGetInstruments (this.extend (request, params));
+        const parsedMarkets = {};
+        let fetchAllMarkets = undefined;
+        [ fetchAllMarkets, params ] = this.handleOptionAndParams (params, 'fetchMarkets', 'fetchAllMarkets', true);
+        if (fetchAllMarkets) {
```

Comment:
why do we need this option if supposedly the single call without currency loads every market similarly to the multiple calls with a currency in each?

Reply:
yes, from my observation we don't need it, but i am not familiar with it's reasons it was coded there, so just for some turbulent period after merge, I allowed users to override it if there was any (unknown to me) differences.

Reply:
we can either remove that right away, or in the end of this month

Reply:
I compared both results (with and without currency) and seems the same, but we can keep it just in case we're missing something

```
{"tick_size_steps":[{"above_price":"0.005","tick_size":"0.0005"}],"quote_currency":"BTC","min_trade_amount":"0.1","counter_currency":"USD","settlement_period":"day","settlement_currency":"BTC","block_trade_tick_size":"0.0001","block_trade_min_trade_amount":"25","block_trade_commission":"0.0003","option_type":"call","creation_timestamp":"1712390460000","instrument_id":"312935","base_currency":"BTC","tick_size":"0.0001","strike":"60000.0","contract_size":"1.0","is_active":true,"expiration_timestamp":"1712649600000","instrument_type":"reversed","instrument_name":"BTC-9APR24-60000-C","taker_commission":"0.0003","maker_commission":"0.0003","kind":"option","rfq":false,"price_index":"btc_usd"}
```

```
{"tick_size_steps":[{"above_price":"0.005","tick_size":"0.0005"}],"quote_currency":"BTC","min_trade_amount":"0.1","counter_currency":"USD","settlement_period":"day","settlement_currency":"BTC","block_trade_tick_size":"0.0001","block_trade_min_trade_amount":"25","block_trade_commission":"0.0003","option_type":"call","creation_timestamp":"1712390460000","instrument_id":"312935","base_currency":"BTC","tick_size":"0.0001","contract_size":"1.0","strike":"60000.0","is_active":true,"expiration_timestamp":"1712649600000","instrument_type":"reversed","instrument_name":"BTC-9APR24-60000-C","taker_commission":"0.0003","maker_commission":"0.0003","kind":"option","rfq":false,"price_index":"btc_usd"}
```

Reply:
btw maybe this was the reason, docs are misleading 
![image](https://github.com/ccxt/ccxt/assets/43336371/ec887f55-2344-4507-a3a4-9a98646d0f6d)

Reply:
yes, exactly

Reply:
the docs said so, yes, i forgot to mention that detail, but it's obvious doc is incorrect